src/skmultiflow/meta/adaptive_random_forest_regressor.py

```diff
@@ -0,0 +1,497 @@
+        Threshold below which a split will be forced to break ties.
+
+    binary_split: bool, optional (default=False)
+        (`RegressionHoeffdingTree` parameter)
+        If True, only allow binary splits.
+
+    stop_mem_management: bool, optional (default=False)
+        (`RegressionHoeffdingTree` parameter)
+        If True, stop growing as soon as memory limit is hit.
+
+    remove_poor_atts: bool, optional (default=False)
+        (`RegressionHoeffdingTree` parameter)
+        If True, disable poor attributes.
+
+    no_preprune: bool, optional (default=False)
+        (`RegressionHoeffdingTree` parameter)
+        If True, disable pre-pruning.
+
+    leaf_prediction: string, optional (default='nba')
+        (`RegressionHoeffdingTree` parameter)
+        Prediction mechanism used at leafs.
+
+        - 'mc' - Majority Class
+        - 'nb' - Naive Bayes
+        - 'nba' - Naive Bayes Adaptive
+
+    nb_threshold: int, optional (default=0)
+        (`RegressionHoeffdingTree` parameter)
+        Number of instances a leaf should observe before allowing Naive Bayes.
+
+    nominal_attributes: list, optional
+        (`RegressionHoeffdingTree` parameter)
+        List of Nominal attributes. If emtpy, then assume that all attributes
+        are numerical.
+
+    learning_ratio_perceptron: float
+        (`RegressionHoeffdingTree` parameter)
+        The learning rate of the perceptron.
+
+    learning_ratio_decay: float
+        (`RegressionHoeffdingTree` parameter)
+        Decay multiplier for the learning rate of the perceptron
+
+    learning_ratio_const: Bool
+        (`RegressionHoeffdingTree` parameter)
+        If False the learning ratio will decay with the number of examples seen
+
+    random_state: int, RandomState instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by `np.random`. Used when leaf_prediction is 'perceptron'.
+
+    Notes
+    -----
+    The 3 most important aspects of Adaptive Random Forest [1]_ are:
+    (1) inducing diversity through re-sampling;
+    (2) inducing diversity through randomly selecting subsets of features for
+        node splits (see skmultiflow.classification.trees.arf_hoeffding_tree);
+    (3) drift detectors per base tree, which cause selective resets
+        in response to drifts.
+    It also allows training background trees, which start training if a warning
+    is detected and replace the active tree if the warning escalates to a drift.
+
+    References
+    ----------
+    .. [1] Gomes, H.M., Bifet, A., Read, J., Barddal, J.P., Enembreck, F.,
+       Pfharinger, B., Holmes, G. and Abdessalem, T., 2017.
+       Adaptive random forests for evolving data stream classification.
+       Machine Learning, 106(9-10), pp.1469-1495. 
+    """
```

Comment:
I think the reference should be updated/extended for the ESANN 2018 paper.

Reply:
@hmgomes What should I do here? Only use the ESANN 2018 reference, or both?

Reply:
> @hmgomes What should I do here? Only use the ESANN 2018 reference, or both?

Use both. The ESANN one is a short paper and some details are missing from it (e.g., some of the parameters are not specified).

Reply:
Ok, I added it, but I'm not sure of the format of the citation, could you please check it?

Reply:
Everything seems fine to me! I'm going to resolve this comment. Thank Alban!